packages/web-frontend/src/containers/EnlargedDialog/EnlargedDialog.js

```diff
@@ -52,39 +45,116 @@ const STATUS = {
   ERROR: 'error',
 };
 
-const EnlargedDialogComponent = ({
-  isDrillDownVisible,
-  onCloseOverlay,
-  viewContent,
-  errorMessage,
-  organisationUnitName,
-  onDrillDown,
-  onSetDateRange,
-  isLoading,
-  isVisible,
-  startDate,
-  endDate,
-  projectCode,
-}) => {
+const getDatesForCurrentLevel = (
+  drillDownLevel,
+  startDateForTopLevel,
+  endDateForTopLevel,
+  drillDownDatesByLevel,
+) => {
+  if (drillDownLevel === 0) {
+    return {
+      startDate: startDateForTopLevel,
+      endDate: endDateForTopLevel,
+    };
+  }
+  return drillDownDatesByLevel?.[drillDownLevel] || {};
+};
+
+const EnlargedDialogComponent = props => {
+  const {
+    onCloseOverlay,
+    contentByLevel,
+    errorMessage,
+    organisationUnitName,
+    onSetDateRange,
+    startDate: startDateForTopLevel,
+    endDate: endDateForTopLevel,
+    isLoading,
+    projectCode,
+    infoViewKey,
+    fetchViewData,
+    drillDownDatesByLevel,
+  } = props;
```

Comment:
Any reason for not simply de-structuring at the top?

Reply:
Oops. It was for console.logging, I'll put it back